app/src/main/java/com/github/onlynotesswent/ui/user/PublicProfile.kt

```diff
@@ -421,16 +428,32 @@ fun FollowUnfollowButton(userViewModel: UserViewModel, otherUserId: String) {
  * @param followerId The ID of the follower to remove.
  */
 @Composable
-fun RemoveFollowerButton(userViewModel: UserViewModel, followerId: String) {
+fun RemoveFollowerButton(
+    userViewModel: UserViewModel,
+    followerId: String,
+    expanded: MutableState<Boolean>?
+) {
   OutlinedButton(
       contentPadding = PaddingValues(horizontal = 10.dp),
       shape = RoundedCornerShape(25),
       modifier = Modifier.testTag("removeFollowerButton--$followerId").width(90.dp),
       onClick = {
         userViewModel.removeFollower(
             followerId,
-            { userViewModel.profileUser.value?.let { userViewModel.refreshProfileUser(it.uid) } },
+            {
+              userViewModel.profileUser.value?.let { userViewModel.refreshProfileUser(it.uid) }
+              // this code will close the bottom sheet after removing the follower (this is a
```

Comment:
it doesn't work every time on my emulator though. this is a minor bug

Reply:
I think instead of using the boolean to fix the problem, you should address the real issue:
the followers/following are loaded once without a state flow from the view model but just to a state (which acts as a variable). as you can see from lines 317 to 325:

```
onClick = {
                    userViewModel.getFollowingFrom(
                        user.value!!.uid,
                        {
                          following.value = it
                          isFollowingMenuShown.value = true
                        },
                        { isFollowingMenuShown.value = false })
                  }
```

to fix this issue, inside bottom sheet composable, try passing a state representing the user the bottom sheets belongs to and calling this code whenever the state changes (with a .let{} for example). This way the remove follower button needs just to call refreshCurrentUser and the bottom sheet reloads itself. Also I noticed there is a mistake, the remove follower button needs to call refreshCurrent user not refresh profile user: this is my mistake you can fix it.